tchannel-core/src/main/java/com/uber/tchannel/handlers/ResponseRouter.java

```diff
@@ -87,8 +88,10 @@ protected void messageReceived(ChannelHandlerContext ctx, RawResponse rawRespons
         ResponsePromise<?> responsePromise = this.messageMap.remove(rawResponse.getId());
```

Comment:
I have a unrelated comment for ResponsePromise, does ResponsePromise fails for a errored Response call ?